### PR TITLE
Cleanup around configuration

### PR DESCRIPTION
**Pull request by skisel-da**
This PR covers the following aspects:
* Remove of unused configuration parameters.
* Create smaller config objects with cohesive parametrisation.
* Removal of redundant transformation layer.
* Preparation for the more generic configuration objects which are not cli-driven.

CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

**Review by remyhaemmerle-da — COMMENTED** [review]
the changes in engine test look good to me.

**Comment by hubert-da** [comment]
kvutils changes look ok 👍🏻

File: ledger/participant-state/kvutils/app/src/test/scala/com/daml/ledger/participant/state/kvutils/app/RunnerSpec.scala [hunk 1]
```diff
@@ -160,12 +160,16 @@ object RunnerSpec {
     ParticipantConfig(
       mode = ParticipantRunMode.Combined,
       participantId = participantId,
-      shardName = None,
+      shardName = Some(UUID.randomUUID().toString),
```
Comment (pbatko-da) [inline ledger/participant-state/kvutils/app/src/test/scala/com/daml/ledger/participant/state/kvutils/app/RunnerSpec.scala:163]: Out of interest, what is this needed for?

File: daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/MinVersionTest.scala [hunk 2]
```diff
@@ -135,22 +132,27 @@ final class MinVersionTest
     port = Port.Dynamic,
     portFile = Some(portfile),
     serverJdbcUrl = ParticipantConfig.defaultIndexJdbcUrl(participantId),
-    indexerConfig = ParticipantIndexerConfig(
-      allowExistingSchema = false
+    indexerConfig = IndexerConfig(
+      participantId = participantId,
+      jdbcUrl = ParticipantConfig.defaultIndexJdbcUrl(participantId),
+      startupMode = IndexerStartupMode.MigrateAndStart(false),
```
Comment (pbatko-da) [inline daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/MinVersionTest.scala:138]: Use named parameter?

**Review by pbatko-da — COMMENTED** [review]
Is this a pure cleanup or is this PR changing/adding new things?

File: ledger/participant-state/kvutils/app/src/test/scala/com/daml/ledger/participant/state/kvutils/app/RunnerSpec.scala (hunk 1 above)
Comment (skisel-da) [reply ledger/participant-state/kvutils/app/src/test/scala/com/daml/ledger/participant/state/kvutils/app/RunnerSpec.scala:163]: so in the previous implementation createMetrics call was overriden to always create new metric registry in `TestConfigProvider` not to clash for Jvm metrics while here i use production version of createMetrics call which supports shards, so I generate random shardName and metrics do not clash.

**Comment by skisel-da** [comment]
> Is this a pure cleanup or is this PR changing/adding new things? Could you summarize main points/goals in the PR's description?

done

**Review by pbatko-da — APPROVED** [review]

**Comment by rgugliel-da** [comment]
Hi!
When about to merge such a PR, can you ping us (Canton) before hand? We need to adapt the config on our side :)

**Comment by skisel-da** [comment]
@rgugliel-da yes, absolutely and sorry for that :) 

heads up: there is a bigger bomb coming at some point: #13629
it's a bigger PR and wont land to main at least for the next 2 weeks
